### ISSUE TITLE
add useWindow  useDocument

### ISSUE THREAD

**Issue by PatrickJS** [BuilderIO/mitosis#398]
I think it's a good idea to add these hooks as each framework has their own way to manage these globals

**Comment by PatrickJS** [comment]
@steve8708 I think we need to add these to Mitosis so that they will be compiled away to the correct way per framework

**Comment by steve8708** [comment]
this is a neat idea. I love compile way features. I haven't used a framework in a little while that had an abstraction for accessing window/document besides just using the global variable names (and in your own code guarding to make sure that would only run in the browser). but maybe some platforms have abstractions here these days like angular?

**Comment by PatrickJS** [comment]
yeah for angular you want to use it's refs to window/document but it's not clear what to do in some other frameworks. In React for example you use it directly but wrap it around a hook.

![Screen Shot 2022-05-23 at 2 38 31 PM](https://user-images.githubusercontent.com/1016365/169885241-b2bf4eb9-f3a0-4f30-b712-b85b700f6950.png)

we may also want to add something like useLayoutEffect

**Comment by steve8708** [comment]
ah yes this is very interesting
